app/src/main/java/com/wafflestudio/snutt2/lib/network/GlobalNetworkEventHandler.kt

```diff
@@ -0,0 +1,35 @@
+package com.wafflestudio.snutt2.lib.network
+
+import android.os.Handler
+import android.os.Looper
+import java.lang.ref.WeakReference
+
+class GlobalNetworkEventHandler {
+    private var mainHandlerRef: WeakReference<(GlobalNetworkEvent) -> Unit>? = null
```

Comment:
android.os.Handler랑 이름이 좀 헷갈리는데 다른 이름 어때(`networkEventHandlerRef`?)
main으로 시작하니 메인스레드 핸들러 같기도 하고

Reply:
```suggestion
    private var mainHandlerRef: WeakReference<(GlobalNetworkEvent) -> Unit> = WeakReference(null)
```
mainHandlerRef 자체는 non-null으로 해서 null check 하나 줄이기

Reply:
WeakReference 쓴 건 싱글턴인 GlobalNetworkEventHandler가 액티비티의 레퍼런스를 물고 늘어져 memory leak 생기는 걸 막기 위함이지? 고수

Reply:
사실 진짜 의도는 따로 있긴 했는데 결국 의도를 달성하지 못했어

Reply:
형이 말한 의도는 2순위 의도였는데... ㅋㅋㅋㅋㅋ 뭐 좋은게 좋은거지 ㅋㅋㅋㅋ

Reply:
뭐야 그럼 1순위의도는 뭔데

Reply:
지금은 mainHandlerRef가 이름인 저거를 var이 아니라 val로 만드는 것

Reply:
아 저거 var이었구나ㅋㅋㅋ 2순위라도 지켜서 좋군

Reply:
이제 생각해보면 val 하고 싶었으면 다른 방법을 찾을걸 그랬나 싶기도 한데... ㅋㅋㅋㅋ
뭐 굿굿입니다

Reply:
<img width="796" alt="image" src="https://github.com/user-attachments/assets/759cd0ce-a5cf-4573-a81f-f82fe0f7d7c3" />

음...?

Reply:
일단 [mainHandlerRef -> networkEventCallbackRef](https://github.com/wafflestudio/snutt-android/pull/432/commits/9bc0bd592ec40ddfd7bdd0462b48328e0339a41c)

Reply:
@eastshine2741

Reply:
오 좋음 굿굿 근데 딴얘긴데 너의 1순위 의도에 제일 맞는 게 lateinit var 아닐까 생각이 방금 들었음
쓰는 곳에서는 ::isInitialized 체크만 한번 해주고
현도 판단에 따라 ㄱㄱ

Reply:
아하 일단 유지해볼게